Sources/wkhtmltopdf/Document+Generate.swift

```diff
@@ -1,16 +1,16 @@
 import Foundation
-import Service
+import NIO
 
 extension Document {
 
-    public func generatePDF(on container: Container) throws -> Future<Data> {
-        let sharedThreadPool = try container.make(BlockingIOThreadPool.self)
-
-        return sharedThreadPool.runIfActive(eventLoop: container.eventLoop) { () -> Data in
+    public func generatePDF(on eventLoop: EventLoop) throws -> EventLoopFuture<Data> {
+        return NIOThreadPool(numberOfThreads: 1).runIfActive(eventLoop: eventLoop) {
```

Comment:
I think the user should pass in their own `NIOThreadPool` here so that it's not set to 1

Reply:
Thanks @mcdappdev! Does it make sense to have a default argument in that case?

```swift
func generatePDF(on threadPool: NIOThreadPool = NIOThreadPool(numberOfThreads: 1), eventLoop: EventLoop)
```

Reply:
Yeah I think that’d be fine!

Reply:
Yeah I think that’d be fine!